### DISCUSSION TITLE
[SOLVED] Can't load session when authorizing

### DISCUSSION THREAD

**Discussion by mqrc81** [xelaj/mtproto#101] [Q&A] [answered]
When using the example code for auth, the session can't be loaded (error below). The session file is recognized (correct path etc.). 

**Do I need to add a session file containing active sessions or is an empty `session.json` file enough?**

As of now, my `session.json` file only contains:
```
{
}
```

Here is my standard auth code:
```
sessionFile := filepath.Join(resourcesFilePath, "session.json")
publicKeysFile := filepath.Join(resourcesFilePath, "tg_public_keys.pem")
appID, _ := strconv.Atoi(os.Getenv("TELEGRAM_ID"))

client, err := telegram.NewClient(telegram.ClientConfig{
        SessionFile:     sessionFile,                
	ServerHost:      os.Getenv("MTPROTO_SERVER"), // address of mtproto proxy server
	PublicKeysFile:  publicKeysFile,              
	AppID:           appID,                       
	AppHash:         os.Getenv("TELEGRAM_HASH"),  
	InitWarnChannel: true,                        
})
```

And here is the error I get:
```
panic: runtime error: index out of range [7] with length 0

goroutine 1 [running]:
encoding/binary.littleEndian.Uint64(...)
	/usr/local/go/src/encoding/binary/binary.go:77
github.com/xelaj/mtproto.(*MTProto).LoadSession(0xc00014aa20, 0xc00014aa20, 0xc000033f70)
	/Users/ms/go/pkg/mod/github.com/xelaj/mtproto@v1.0.0/session_save.go:47 +0x165
github.com/xelaj/mtproto.NewMTProto(0xc00003c5e0, 0x16, 0xc00003684f, 0x14, 0xc000033960, 0x0, 0x0, 0x13)
	/Users/ms/go/pkg/mod/github.com/xelaj/mtproto@v1.0.0/mtproto.go:97 +0x69
github.com/xelaj/mtproto/telegram.NewClient(0xc00003c5e0, 0x16, 0xc00003684f, 0x14, 0xc00003c600, 0x1c, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/ms/go/pkg/mod/github.com/xelaj/mtproto@v1.0.0/telegram/common.go:70 +0x2c5
main.main()
	/Users/ms/Projects/crypto-prophet/main.go:31 +0x3b8
```

**Comment by mqrc81** [comment, marked as answer]
Apparently it created a session file itself called "tg_session.json" (what I initially called the file) which I missed for a whole week... fml

**Reply by Jimweb3** [reply]
HI,
I have the same question, can you explain it in more detail, thank you very much

**Reply by mqrc81** [reply]
Whatever you name the sessions-file the first time you run this code (as long as everything else is correct), there will be a file created in that location. So you don't have to preemptively create a sessions-file.

Say you run the above code with the filename being "sessions_tg.json", then it will create this file & from there on out you should always point to that file.

So if you get an error, the filepath you specified might either be taken by a file you created or you've already automatically created such a file by running the script correctly previously and are now specifying a different filepath. (Of course there might also be errors not having to do with the sessions file.)

**Reply by quenbyako** [reply]
I'm sorry for long (well, **too long**) response, but... 

Try to think a little and understand why this does not work: mtproto thinks that the session exists (After all, there is a session file! It is logical ...), however, there is no data required for work / they are incorrect (empty salt field, session key, etc. etc.)

Let's think about it in terms of handling a situation like this: how to respond to it? Lack of session? Unlikely, because the file exists. Broken session? Now it is. Do I need to recreate the session? What if you have some important data there, but I, as a library developer, cannot allow it to look into your files and see what data is there?

**Reply by quenbyako** [reply]
Also, in last commit's i've created additional methods for getting session data (well, the whole new interface, with default value as file storage), so try to use it, may'be it will help you :)
